src/server/action_handlers/revere-signup.js

```diff
@@ -22,76 +25,105 @@ export async function available(organizationId) {
-      .select('campaign_contact.cell', 'campaign_contact.first_name', 'campaign_contact.last_name')
+      .select('campaign_contact.cell', 'campaign_contact.first_name', 'campaign_contact.last_name', 'campaign_contact.custom_fields')
+
   const contact = (contactRes.length ? contactRes[0] : {})
+  const customFields = JSON.parse(contact.custom_fields)
+  const mobileFlowId = (customFields.revere_signup_flow ? customFields.revere_signup_flow : defaultMobileFlowId)
   const contactCell = contact.cell.substring(1)
 
-  const options = {
-    method: 'POST',
-    url: 'https://mobile.reverehq.com/api/v1/messaging/sendContent',
-    headers: {
-      accept: 'application/json',
-      'content-type': 'application/json',
-      Authorization: mobileApiKey
-    },
-    body: {
-      msisdns: [`00${contactCell}`],
-      mobileFlow: `${mobileFlowId}`
-    },
-    json: true
-  }
-
-  return request(options, (error, response) => {
-    if (error) throw new Error(error)
-    if (response.statusCode === 204 && akAddUserUrl) {
-      const userData = {
-        email: contactCell + '-smssubscriber@example.com',
-        first_name: contact.first_name,
-        last_name: contact.last_name,
-        sms_subscribed: true,
-        action_mobilesubscribe: true,
-        suppress_subscribe: true,
-        phone: [contactCell],
-        phone_type: 'mobile',
-        source: 'spoke-signup'
+  if (sqsUrl) {
+    const msg = {
+      payload: {
+        cell: `${contactCell}`,
+        mobile_flow_id: `${mobileFlowId}`,
+        source: 'spoke'
       }
+    }
 
-      request.post({
-        url: akAddUserUrl,
-        headers: {
-          accept: 'application/json',
-          'content-type': 'application/json'
-        },
-        form: userData
-      }, (errorResponse, httpResponse) => {
-        if (errorResponse) throw new Error(errorResponse)
-        if (httpResponse.statusCode === 201) {
-          request.post({
-            url: akAddPhoneUrl,
-            headers: {
-              accept: 'application/json',
-              'content-type': 'application/json'
-            },
-            form: {
-              user: httpResponse.headers.location,
-              phone: contactCell,
-              type: 'mobile'
-            }
-          }, (lastError, lastResponse) => {
-            if (lastError) throw new Error(lastError)
-            if (lastResponse.statusCode === 201) {
-              return
-            }
-          })
-        }
-      })
-    } else {
-      return
+    const sqsParams = {
+      MessageBody: JSON.stringify(msg),
+      QueueUrl: sqsUrl
     }
-  })
+
+    sqs.sendMessage(sqsParams, (err, data) => {
+      if (err) {
+        console.log('Error sending message to queue', err);
+      }
+      console.log('Sent message to queue with data:', data);
+    })
+  } else {
+    const options = {
+      method: 'POST',
+      url: 'https://mobile.reverehq.com/api/v1/messaging/sendContent',
+      headers: {
```

Comment:
should the revere URL be an env variable?

Reply:
I would make it an env variable anyway, unless this is the only place where it will be used. Otherwise updating or changing the URL is going to be a big pain 🤕